From 88d87b1ac2b5f8a1aecc339f92a9fca57d366bcc Mon Sep 17 00:00:00 2001
From: Igor Opaniuk <igor.opaniuk@toradex.com>
Date: Wed, 19 Jun 2019 11:47:06 +0300
Subject: [PATCH 2/6] ARM: dts: colibri_imx7: Fix lcdif node definition

Fix lcdif DT node and make it conform to the structure defined in the
Linux devicetree bindings [1]. Currently there is support only for
old style lcdif node definitions.

[1] https://www.kernel.org/doc/Documentation/devicetree/bindings/display/mxsfb.txt

Signed-off-by: Igor Opaniuk <igor.opaniuk@toradex.com>
---
 arch/arm/dts/imx7-colibri.dtsi | 47 +++++++++++++++++++++++-------------------
 1 file changed, 26 insertions(+), 21 deletions(-)

diff --git a/arch/arm/dts/imx7-colibri.dtsi b/arch/arm/dts/imx7-colibri.dtsi
index 81717c233d..308e0b2a63 100644
--- a/arch/arm/dts/imx7-colibri.dtsi
+++ b/arch/arm/dts/imx7-colibri.dtsi
@@ -113,29 +113,34 @@
 };
 
 &lcdif {
-	u-boot,dm-pre-reloc;
 	status = "okay";
+	display = <&display0>;
+	u-boot,dm-pre-reloc;
 
-	display-timings {
-		native-mode = <&timing_vga>;
-
-		/* Standard VGA timing */
-		timing_vga: 640x480 {
-			u-boot,dm-pre-reloc;
-			clock-frequency = <25175000>;
-			hactive = <640>;
-			vactive = <480>;
-			hback-porch = <48>;
-			hfront-porch = <16>;
-			vback-porch = <33>;
-			vfront-porch = <10>;
-			hsync-len = <96>;
-			vsync-len = <2>;
-
-			de-active = <1>;
-			hsync-active = <0>;
-			vsync-active = <0>;
-			pixelclk-active = <0>;
+	display0: display0 {
+		bits-per-pixel = <18>;
+		bus-width = <24>;
+		status = "okay";
+
+		display-timings {
+			native-mode = <&timing_vga>;
+			timing_vga: 640x480 {
+				u-boot,dm-pre-reloc;
+				clock-frequency = <25175000>;
+				hactive = <640>;
+				vactive = <480>;
+				hback-porch = <48>;
+				hfront-porch = <16>;
+				vback-porch = <33>;
+				vfront-porch = <10>;
+				hsync-len = <96>;
+				vsync-len = <2>;
+
+				de-active = <1>;
+				hsync-active = <0>;
+				vsync-active = <0>;
+				pixelclk-active = <0>;
+			};
 		};
 	};
 };
-- 
2.13.6

